controllers.js

```diff
@@ -17,4 +18,18 @@ exports.getArticle = (req, res, next) => {
         res.status(200).send({articles});
     })
     .catch(next);
+}
+
+exports.getArticleById = (req, res, next) => {
+    const article_id = req.params.article_id;
+    
+    getArticleByIdModels(article_id)
+    .then( (article) => {
+        if (article.length !== 0) {
+            res.status(200).send({article: article[0]});
+        } else {
+            next();

```

Comment:
Line 31 is doing something really interesting that I didn't spot before. When we invoke next with nothing we are telling next to go to the next middleware (not specifically error middleware but any middleware). In this case, it's going to the app.js and moving onto the next line after the GET route we're on, line 17 which throws a not found error for bad paths. 

So in this case, we are throwing a 404 but not the right 404. 

Instead of line 31, you need to throw your own error in this moment, a custom error. This will be done using Promise.reject. The idea is that the rejection will be caught by the catch on line 34 and then sent to actual error handling middleware. 

To support this functionality I actually need the if else statement to be moved from line 28 in this file to line 21 in your model file. All 'data touching' should be done in the model so the controller is only concerned about sending responses or catching errors.

Reply:
The res.status.send should stay in the controller though - so all you're doing in the model is deciding whether to return the article or throw a error